### DISCUSSION TITLE
Pass messages from host to iframe embedded code-server

### DISCUSSION THREAD

**Discussion by ksdme** [coder/code-server#5186] [Q&A]
Hey people,

I have an instance of code-server running and it is embedded in an iframe on a page with several buttons. I am trying to figure out a way to send a message/event to my custom extension in code-server from the host page when one of the buttons is clicked.

I understand that by hacking the entrypoint file of the frontend (`src/vs/code/browser/workbench/workbench.html`?) I can pass a message from the host frame to the vscode iframe using postMessage, but I haven't been able to find a way to propagate it from there to my extension. https://github.com/microsoft/vscode/pull/138054 looks relevant but my question still stands even if I switch to web extensions.

**Comment by jsjoeio** [comment]
Hmm...seems like that PR is relevant. Besides that, I see [this](https://code.visualstudio.com/api/references/vscode-api) in the docs, but probably not directly relevant. 

What is missing from the PR you posted?

**Comment by RahulBadenkal** [comment]
Hi @ksdme did you find a solution to your problem? I am in the same boat and just wanted to know if you were able to find a way to do it

**Reply by ksdme** [reply]
Sorry for the late response. But, I think the way I solved this at the time was by writing an extension that exposed a REST interface. The extension would basically take commands over REST and push them into vscode events handling infrastructure. I could not solve it using simple cross frame messaging.

**Reply by rahulwecp** [reply]
Hi,
Thanks for the response!
But how would you send message from vs code to the parent iframe?

**Reply by yeldhomathew** [reply]
@ksdme Is it possible to share the extension you developed. Just want to check if it is possible to adapt in our use-case

**Reply by IslombekHasan** [reply]
I think one way to do this would be using WebSockets, or even maybe a shared DB like Firebase RealTime Database, pushing/pulling messages from there based on an identifier. 

I'm going through the same thing right now and that's the way I see it for now.
